elastic/eui
# [Emotion] Convert EuiDataGrid (Part 1)

## Summary

Part of #6394. I'm splitting up EuiDataGrid into smaller chunks for ease of review/QA, but as always, I still recommend [following along by commit](https://github.com/elastic/eui/pull/7998/commits).

Converts the following miscellaenous parts of EuiDataGrid to Emotion:

- EuiDataGrid wrapper(s) + grid body
- Full screen CSS
- Scrollbar overlay/styles
- Pagination

## QA

- Scrollbar testing
  - Turn your Mac system's `Show scroll bars` setting to `Always`
  - Go to https://eui.elastic.co/pr_7998/storybook/?path=/story/tabular-content-euidatagrid--virtualization
  - [x] Confirm the scrollbars are afforded for in the datagrid on all browsers (FF, Safari, Edge)
  - Turn off always-shown system scrollbars (i.e., `Automatically`)
  - [x] Confirm the scrollbars correctly show as overlays and the custom scrollbar CSS is no longer rendered
- Fullscreen testing
  - Go to https://eui.elastic.co/pr_7998/storybook/?path=/story/tabular-content-euidatagrid-fullscreenvrt--full-screen-with-header
  - [x] Confirm that the EuiHeader does not show in fullscreen mode (closing fullscreen mode should show the fixed header)
  - Go to https://eui.elastic.co/pr_7998/storybook/?path=/story/tabular-content-euidatagrid-fullscreenvrt--full-screen-with-flyout
  - [x] Confirm that the EuiFlyout does not account for the fixed EuiHeader when in fullscreen mode
  - [x] Close the flyout and then close datagrid fullscreen mode, and then click the cell button to toggle the flyout. Confirm the EuiHeader is correctly accounted for by the flyout when not in fullscreen mode
- EuiDataGridBody
  - [x] [Custom body renderer example](https://eui.elastic.co/pr_7998/#/tabular-content/data-grid-advanced#custom-body-renderer) not look visually broken/regressed

### General checklist

- Browser QA
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA - N/A
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
